contracts/misc/AutoYieldApe.sol

```diff
@@ -0,0 +1,357 @@
+        _harvest(sqrtPriceLimitX96);
+    }
+
+    /// @inheritdoc IAutoYieldApe
+    function yieldAmount(address account)
+        public
+        view
+        override
+        returns (uint256)
+    {
+        uint256 pendingYield = _userPendingYield[account];
+        uint256 indexDiff = _currentYieldIndex - _userYieldIndex[account];
+        uint256 userBalance = balanceOf(account);
+        if (indexDiff > 0 && userBalance > 0) {
+            uint256 rewardDiff = (userBalance * indexDiff) / RAY;
+            pendingYield += rewardDiff;
+        }
+
+        if (pendingYield > 0) {
+            uint256 liquidityIndex = _lendingPool.getReserveNormalizedIncome(
+                _yieldUnderlying
+            );
+            pendingYield = pendingYield.rayMul(liquidityIndex);
+        }
+
+        return pendingYield;
+    }
+
+    /// @inheritdoc IYieldInfo
+    function yieldIndex() external view override returns (uint256) {
+        return _currentYieldIndex;
+    }
+
+    /// @inheritdoc IYieldInfo
+    function yieldToken() external view override returns (address) {
+        return address(_yieldToken);
+    }
+
+    /// @inheritdoc IYieldInfo
+    function yieldInfo()
+        external
+        view
+        override
+        returns (
+            address,
+            address,
+            uint256
+        )
+    {
+        return (_yieldUnderlying, _yieldToken, _currentYieldIndex);
+    }
+
+    function setHarvestOperator(address _harvestOperator) external onlyOwner {
+        require(_harvestOperator != address(0), "zero address");
+        address oldOperator = harvestOperator;
+        if (oldOperator != _harvestOperator) {
+            harvestOperator = _harvestOperator;
+            emit HarvestOperatorUpdated(oldOperator, _harvestOperator);
+        }
+    }
+
+    function setHarvestFee(uint256 _harvestFee) external onlyOwner {
+        require(
+            _harvestFee < PercentageMath.HALF_PERCENTAGE_FACTOR,
+            "Fee Too High"
+        );
+        uint256 oldValue = harvestFee;
+        if (oldValue != _harvestFee) {
+            harvestFee = _harvestFee;
+            emit HarvestFeeUpdated(oldValue, _harvestFee);
+        }
+    }
+
+    function rescueERC20(
+        address token,
+        address to,
+        uint256 amount
+    ) external onlyOwner {
+        require(token != address(_yieldToken), "cannot rescue yield token");
+        IERC20(token).safeTransfer(to, amount);
+        emit RescueERC20(token, to, amount);
+    }
+
+    function _withdraw(uint256 amount) internal {
+        require(amount > 0, "zero amount");
+
+        _updateYieldIndex(msg.sender);
+        _burn(msg.sender, amount);
+
+        _apeStaking.withdrawSelfApeCoin(amount);
+        IERC20(_apeCoin).safeTransfer(msg.sender, amount);
+
+        emit Redeem(msg.sender, amount);
+    }
+
+    function _claim() internal {
+        uint256 pendingYield = yieldAmount(msg.sender);
```

Comment:
the naming is confusing, pendingYield is the pUSD share, but here it reflect pUSD share * liquidityIndex